`Error: rpc error: code = Unavailable desc = transport is closing` in release v1.2.0

### Provider Version
1.2.0

### Terraform Version
v0.12.24

### Affected Resource(s)
- helm_repository

### Terraform Configuration Files
```hcl
locals {
    local_storage_name = "xyz-local"
    helm_repo_url = var.helm_repo_url == "" ? "https://some.url" : var.helm_repo_url
    shared_storage_name = "xyz-shared"
    tls_secret_name = "xyz-tls"
}

data "helm_repository" "bob" {
    name = "bob"
    url  = local.helm_repo_url
    username = var.helm_repo_username
    password = var.helm_repo_password
}

resource "helm_release" "joe" {
    name = "xyz"
    repository = data.helm_repository.bob.metadata[0].name
    chart      = var.chart
    version = var.joe_version
}
```

### Debug Output
```
Warning: This resource is deprecated and will be removed in the next major version.
  on ../../modules/xyz/main.tf line 8, in data "helm_repository" "xyz": 
   8: data "helm_repository" "xyz" { 

Error: rpc error: code = Unavailable desc = transport is closing
```

### Panic Output
```
panic: runtime error: index out of range [1] with length 1
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: goroutine 100 [running]:
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resultToError(0xc00100e140, 0xc000695420, 0x1)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1036 +0x2e4
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.lintChart(0xc00017d4a0, 0xc000c19760, 0x1e, 0xc00031c1b0, 0xc000c2d2f0, 0x1e, 0x0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1022 +0x106
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resourceReleaseValidate(0x2448600, 0xc000a7c540, 0x1e57f20, 0xc00017d4a0, 0xc00031c1b0, 0xc00031c870, 0xc00089a0a0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1010 +0x181
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resourceDiff(0xc000a7c540, 0x1e57f20, 0xc00017d4a0, 0xc000c57d40, 0xc000a7c540)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:651 +0x175
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.schemaMap.Diff(0xc000157590, 0xc000a5d130, 0xc000459350, 0x2150258, 0x1e57f20, 0xc00017d4a0, 0x0, 0x249be00, 0xc00078df30, 0x0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/schema.go:518 +0xac2
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).simpleDiff(0xc000a383f0, 0xc000a5d130, 0xc000459350, 0x1e57f20, 0xc00017d4a0, 0xc000459301, 0xc000707840, 0x40d49d)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:356 +0x85
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).SimpleDiff(0xc0003fe000, 0xc000707a28, 0xc000a5d130, 0xc000459350, 0xc0002ce450, 0xc000459350, 0x0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:321 +0x99
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).PlanResourceChange(0xc000a1c018, 0x249a800, 0xc0002ce3f0, 0xc0008e6000, 0xc000a1c018, 0xc0002ce3f0, 0xc000a5bb30)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:633 +0x793
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x1fc6c40, 0xc000a1c018, 0x249a800, 0xc0002ce3f0, 0xc00031f740, 0x0, 0x249a800, 0xc0002ce3f0, 0xc000a68500, 0x495)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3287 +0x217
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000455080, 0x24c2d00, 0xc000548c00, 0xc000830400, 0xc000680990, 0x3506608, 0x0, 0x0, 0x0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:1024 +0x4f4
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc000455080, 0x24c2d00, 0xc000548c00, 0xc000830400, 0x0)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:1313 +0xd97
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0007b8df0, 0xc000455080, 0x24c2d00, 0xc000548c00, 0xc000830400)
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:722 +0xbb
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-05-06T19:35:11.500Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:720 +0xa1
2020/05/06 19:35:11 [ERROR] module.gradient_aws.module.gradient_processing: eval: *terraform.EvalDiff, err: rpc error: code = Unavailable desc = transport is closing
2020-05-06T19:35:11.505Z [DEBUG] plugin: plugin process exited: path=/root/project/gradient-aws/tests/.terraform/plugins/linux_amd64/terraform-provider-helm_v1.2.0_x4 pid=1079 error="exit status 2"
2020/05/06 19:35:11 [ERROR] module.gradient_aws.module.gradient_processing: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/05/06 19:35:11 [TRACE] [walkPlan] Exiting eval tree: module.gradient_aws.module.gradient_processing.helm_release.gradient_processing
2020/05/06 19:35:11 [TRACE] vertex "module.gradient_aws.module.gradient_processing.helm_release.gradient_processing": visit complete
2020/05/06 19:35:11 [TRACE] vertex "module.gradient_aws.module.gradient_processing.helm_release.gradient_processing": dynamic subgraph encountered errors
2020/05/06 19:35:11 [TRACE] vertex "module.gradient_aws.module.gradient_processing.helm_release.gradient_processing": visit complete
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "module.gradient_aws.provider.helm.gradient (close)" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "module.gradient_aws.module.gradient_processing.data.kubernetes_service.traefik" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "module.gradient_aws.provider.kubernetes.gradient (close)" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "module.gradient_aws.module.gradient_processing.output.traefik_service" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "module.gradient_aws.output.elb_hostname" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/05/06 19:35:11 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/05/06 19:35:11 [INFO] backend/local: plan operation completed
2020/05/06 19:35:11 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2020/05/06 19:35:11 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2020-05-06T19:35:11.507Z [DEBUG] plugin: plugin exited
2020-05-06T19:35:11.510Z [DEBUG] plugin: plugin process exited: path=/root/project/gradient-aws/tests/.terraform/plugins/linux_amd64/terraform-provider-kubernetes_v1.11.1_x4 pid=1108
2020-05-06T19:35:11.510Z [DEBUG] plugin: plugin exited
```

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
Running this as a CircleCI job:
```
Build-agent version 1.0.31716-0fe2705f (2020-05-06T15:46:07+0000)
Docker Engine Version: 18.09.6
Kernel Version: Linux d3e0348b0c22 4.15.0-1052-aws #54-Ubuntu SMP Tue Oct 1 15:43:26 UTC 2019 x86_64 Linux
Starting container hashicorp/terraform:0.12.24
  image is cached as hashicorp/terraform:0.12.24, but refreshing...
0.12.24: Pulling from hashicorp/terraform
Digest: sha256:53fb1c0a78c8bb91c4a855c1b352ea7928f6fa65f8080dc7a845e240dd2a9bee
Status: Image is up to date for hashicorp/terraform:0.12.24
  using image hashicorp/terraform@sha256:53fb1c0a78c8bb91c4a855c1b352ea7928f6fa65f8080dc7a845e240dd2a9bee
```

### References
I notice at https://github.com/terraform-providers/terraform-provider-aws/issues/11420 that they determined that they needed to introduce greater defensiveness on the input value, but I don't know if this is at all related, just a similar error. I'm mid-debugging, so I'll update if I find out more.

Same here

```
Acquiring state lock. This may take a few moments...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


------------------------------------------------------------------------

Warning: "set_string": [DEPRECATED] This argument is deprecated and will be removed in the next major version. Use `set` argument with `type` equals to `string`

  on main.tf line 1, in resource "helm_release" "mercuryapp":
   1: resource "helm_release" "mercuryapp" {



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: index out of range [1] with length 1
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: goroutine 33 [running]:
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resultToError(0xc000ad7980, 0xc00094b420, 0x1)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1036 +0x2e4
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.lintChart(0xc0002473b0, 0xc0003d6e20, 0x15, 0xc00013eb40, 0xc000ab6030, 0x15, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1022 +0x106
```

crash.log
```
2020-05-07T03:36:40.800Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: panic: runtime error: index out of range [1] with length 1
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: goroutine 33 [running]:
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resultToError(0xc000ad7980, 0xc00094b420, 0x1)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1036 +0x2e4
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.lintChart(0xc0002473b0, 0xc0003d6e20, 0x15, 0xc00013eb40, 0xc000ab6030, 0x15, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1022 +0x106
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resourceReleaseValidate(0x2448600, 0xc000a38f40, 0x1e57f20, 0xc0002473b0, 0xc00013eb40, 0xc00013e3f0, 0xc0004a8780)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:1010 +0x181
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/terraform-providers/terraform-provider-helm/helm.resourceDiff(0xc000a38f40, 0x1e57f20, 0xc0002473b0, 0xc0009936a0, 0xc000a38f40)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/helm/resource_release.go:651 +0x175
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.schemaMap.Diff(0xc0004038f0, 0xc000786820, 0xc00097cc60, 0x2150258, 0x1e57f20, 0xc0002473b0, 0x1ca4100, 0x249be00, 0xffffffffffffffff, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/schema.go:518 +0xac2
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).simpleDiff(0xc00039f3b0, 0xc000786820, 0xc00097cc60, 0x1e57f20, 0xc0002473b0, 0xc00097cc01, 0xc00094b840, 0x40d49d)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:356 +0x85
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).SimpleDiff(0xc000675980, 0xc00094ba28, 0xc000786820, 0xc00097cc60, 0xc000914d50, 0xc00097cc60, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:321 +0x99
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).PlanResourceChange(0xc00000ea10, 0x249a800, 0xc000914cf0, 0xc0007da0e0, 0xc00000ea10, 0xc000914cf0, 0xc000889b30)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:633 +0x793
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x1fc6c40, 0xc00000ea10, 0x249a800, 0xc000914cf0, 0xc00091a120, 0x0, 0x249a800, 0xc000914cf0, 0xc00091d300, 0x105f)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3287 +0x217
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000275200, 0x24c2d00, 0xc000275e00, 0xc00011e700, 0xc000602540, 0x3506608, 0x0, 0x0, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:1024 +0x4f4
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc000275200, 0x24c2d00, 0xc000275e00, 0xc00011e700, 0x0)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:1313 +0xd97
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00048e870, 0xc000275200, 0x24c2d00, 0xc000275e00, 0xc00011e700)
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:722 +0xbb
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-05-07T03:36:40.804Z [DEBUG] plugin.terraform-provider-helm_v1.2.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-helm/vendor/google.golang.org/grpc/server.go:720 +0xa1
2020-05-07T03:36:40.807Z [DEBUG] plugin: plugin process exited: path=/home/azuredevopsuser/agent/agent-1/_work/1/s/build/terraform-mercuryapp/.terraform/plugins/linux_amd64/terraform-provider-helm_v1.2.0_x4 pid=25560 error="exit status 2"
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalDiff, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [TRACE] [walkPlan] Exiting eval tree: helm_release.provider0
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalDiff, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [TRACE] [walkPlan] Exiting eval tree: helm_release.provider1
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider1": visit complete
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalDiff, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider1": dynamic subgraph encountered errors
2020/05/07 03:36:40 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/05/07 03:36:40 [TRACE] [walkPlan] Exiting eval tree: helm_release.mercuryapp
2020/05/07 03:36:40 [TRACE] vertex "helm_release.mercuryapp": visit complete
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider0": visit complete
2020/05/07 03:36:40 [TRACE] vertex "helm_release.mercuryapp": dynamic subgraph encountered errors
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider0": dynamic subgraph encountered errors
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider0": visit complete
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "output.provider0_release_name" errored, so skipping
2020/05/07 03:36:40 [TRACE] vertex "helm_release.provider1": visit complete
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "output.provider1_release_name" errored, so skipping
2020/05/07 03:36:40 [TRACE] vertex "helm_release.mercuryapp": visit complete
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "provider.helm (close)" errored, so skipping
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "output.release_name" errored, so skipping
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/05/07 03:36:40 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/05/07 03:36:40 [INFO] backend/local: plan operation completed
2020-05-07T03:36:41.086Z [DEBUG] plugin: plugin exited
```

--- a/helm/resource_release.go
+++ b/helm/resource_release.go
@@ -1027,14 +1027,15 @@ func resultToError(r *action.LintResult) error {
 		return nil
 	}
 
-	errMsg := make([]string, len(r.Errors))
-	for i, msg := range r.Messages {
-		if msg.Err == nil {
-			continue
+	messages := []string{}
+	for _, msg := range r.Messages {
+		for _, err := range r.Errors {
+			if err == msg.Err {
+				messages = append(messages, fmt.Sprintf("%s: %s", msg.Path, msg.Err))
+				break
+			}
 		}
-
-		errMsg[i] = fmt.Sprintf("%s: %s", msg.Path, msg.Err)
 	}
 
-	return fmt.Errorf("malformed chart or values: \n\t%s", strings.Join(errMsg, "\n\t"))
+	return fmt.Errorf("malformed chart or values: \n\t%s", strings.Join(messages, "\n\t"))
 }

--- a/helm/resource_release_test.go
+++ b/helm/resource_release_test.go
@@ -1078,7 +1078,7 @@ func testAccHelmReleaseConfigPostrender(resource, ns, name, binaryPath string) s
 	`, resource, name, ns, binaryPath)
 }
 
-func TestAccResourceRelease_LintFail(t *testing.T) {
+func TestAccResourceRelease_LintFailValues(t *testing.T) {
 	namespace := fmt.Sprintf("%s-%s", testNamespace, acctest.RandString(10))
 	defer deleteNamespace(t, namespace)
 
@@ -1101,6 +1101,30 @@ func TestAccResourceRelease_LintFail(t *testing.T) {
 			Config:             broken,
 			PlanOnly:           true,
 			ExpectError:        regexp.MustCompile("malformed chart or values"),
+			ExpectNonEmptyPlan: true,
+		}},
+	})
+}
+
+func TestAccResourceRelease_LintFailChart(t *testing.T) {
+	namespace := fmt.Sprintf("%s-%s", testNamespace, acctest.RandString(10))
+	defer deleteNamespace(t, namespace)
+
+	broken := fmt.Sprintf(`
+	resource "helm_release" "test" {
+		name        = "foo"
+		namespace   = %q
+		chart       = "./test-fixtures/charts/broken-chart"
+	}`, namespace)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, namespace) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckHelmReleaseDestroy(namespace),
+		Steps: []resource.TestStep{{
+			Config:             broken,
+			PlanOnly:           true,
+			ExpectError:        regexp.MustCompile(`function "BAD_FUNCTION" not defined`),
 			ExpectNonEmptyPlan: true,
 		}},
 	})

--- a/helm/test-fixtures/charts/broken-chart/Chart.yaml
+++ b/helm/test-fixtures/charts/broken-chart/Chart.yaml
@@ -1,0 +1,6 @@
+apiVersion: v2
+name: broken-chart
+description: A broken chart for testing the Helm provider
+type: application
+version: 1.2.3
+appVersion: 1.2.3

--- a/helm/test-fixtures/charts/broken-chart/templates/service.yaml
+++ b/helm/test-fixtures/charts/broken-chart/templates/service.yaml
@@ -1,0 +1,13 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: test
+spec:
+  type: NodePort
+  ports:
+    - port: 80
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    {{- BAD_FUNCTION }}

